src/tests/suite/dfs_unit_test.c

```diff
@@ -691,6 +692,78 @@ dfs_test_io_error_code(void **state)
 	assert_int_equal(rc, 0);
 }
 
+int dfs_test_rc[DFS_TEST_MAX_THREAD_NR];
+
+static void *
+dfs_test_mkdir_thread(void *arg)
+{
+	struct dfs_test_thread_arg	*targ = arg;
+	int				rc;
+
+	pthread_barrier_wait(targ->barrier);
+	rc = dfs_mkdir(dfs_mt, NULL, targ->name, S_IFDIR, OC_S1);
+	print_message("dfs_test_read_thread %d, dfs_mkdir rc %d.\n",
+		      targ->thread_idx, rc);
+	dfs_test_rc[targ->thread_idx] = rc;
+	pthread_exit(NULL);
+}
+
+static void
+dfs_test_mt_mkdir(void **state)
+{
+	test_arg_t		*arg = *state;
+	char			name[16];
+	pthread_barrier_t	barrier;
+	int			i, one_success;
+	int			rc;
+
+	MPI_Barrier(MPI_COMM_WORLD);
+
+	sprintf(name, "MTA_dir_%d\n", arg->myrank);
+
+	/* usr barrier to all threads start at the same time and start
+	 * concurrent test.
+	 */
+	pthread_barrier_init(&barrier, NULL, dfs_test_thread_nr + 1);
+	for (i = 0; i < dfs_test_thread_nr; i++) {
+		dfs_test_targ[i].thread_idx = i;
+		dfs_test_targ[i].name = name;
+		dfs_test_targ[i].barrier = &barrier;
+		rc = pthread_create(&dfs_test_tid[i], NULL,
+				    dfs_test_mkdir_thread, &dfs_test_targ[i]);
+		assert_int_equal(rc, 0);
```

Comment:
wouldn't be the first test to have this problem but we leak resources if this fires